cmd/clusterctl/cmd/config_cluster.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+)
+
+type configClusterOptions struct {
+	kubeconfig             string
+	flavor                 string
+	bootstrapProvider      string
+	infrastructureProvider string
+
+	targetNamespace   string
+	kubernetesVersion string
+	controlplaneCount int
+	workerCount       int
+}
+
+var cc = &configClusterOptions{}
+
+var configClusterClusterCmd = &cobra.Command{
+	Use:   "cluster",
+	Short: "Generate templates for creating a cluster API workload clusters",
+	Long: LongDesc(`
+		clusterctl ships with a list of well know providers; if necessary, edit
+		the .clusterctl file to add new provider configurations or to customize existing ones.
+		
+		Each provider configuration links to a repository, and clusterctl will fetch the template
+		from creating a new cluster from there.`),
+
+	Example: Examples(`
+		# Generates a yaml file for for creating a cluster API workload clusters using
+		# default infrastructure provider and default bootstrap provider installed in the cluster.
+		clusterctl config cluster my-cluster
+ 
+		# Generates a yaml file for for creating a cluster API workload clusters using
+		# specified infrastructure and bootstrap provider
+		clusterctl config cluster my-cluster --infrastructure=aws --bootstrap=kubeadm
+ 
+		# Generates a yaml file for for creating a cluster API workload clusters using
+		# specified version of the AWS infrastructure provider
+		clusterctl config cluster my-cluster --infrastructure=aws:v0.4.1
```

Comment:
The default behavior use introspection, but it can't work in all the conditions:
- `clusterctl config cluster my-cluster` (without specifying any provider) works if there is only one infra/bootstrap provider installed in the cluster; if this is not the case, you have to specify which provider to use with `--infrastructure` or `--bootstrap=` 
- `clusterctl config cluster my-cluster --infrastructure=aws` (without specifying the version) works if there is only one version installed for a specific provider ; if this is not the case, you have to specify which provider/version to use with `--infrastructure aws:v04.5.1`(*)

(*) I would like to rule this last option out of the table, but this is strictly linked to what will be decided for https://github.com/kubernetes-sigs/cluster-api/issues/1631

Reply:
sgtm, thanks